baggage-processor/src/main/java/io/opentelemetry/contrib/baggage/processor/BaggageSpanProcessorCustomizer.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.contrib.baggage.processor;
+
+import io.opentelemetry.sdk.autoconfigure.spi.AutoConfigurationCustomizer;
+import io.opentelemetry.sdk.autoconfigure.spi.AutoConfigurationCustomizerProvider;
+import io.opentelemetry.sdk.autoconfigure.spi.ConfigProperties;
+import io.opentelemetry.sdk.trace.SdkTracerProviderBuilder;
+import java.util.List;
+
+public class BaggageSpanProcessorCustomizer implements AutoConfigurationCustomizerProvider {
+  @Override
+  public void customize(AutoConfigurationCustomizer autoConfigurationCustomizer) {
+    autoConfigurationCustomizer.addTracerProviderCustomizer(
```

Comment:
For a moment I thought this wouldn't work because this customizers are [applied after the batch exporter is registered](https://github.com/open-telemetry/opentelemetry-java/blob/788347e262b34ccf287a5b32a9976eb43e170c5f/sdk-extensions/autoconfigure/src/main/java/io/opentelemetry/sdk/autoconfigure/AutoConfiguredOpenTelemetrySdkBuilder.java#L473-L475), but then I realized that order of registration doesn't matter for span processors which want to take advantage of modifying the span on start.

We would need to make changes in the SDK if this was a `BaggageLogRecordProcessor`: Either providing a hook to customize the logger provider before batch processor is added, or by extending `SdkLoggerProviderBuilder` with a method to insert a processor at the front of the queue.

Reply:
> Either providing a hook to customize the logger provider before batch processor is added

I think this is a great idea - would make sense for all signals

BTW, I've made the test end-to-end to cover that "onStast" is actually called